media/windows/ECC-MQTT-IoT/ECC_MQTT_IoT_SQLite.py

```diff
@@ -101,16 +124,149 @@
+            os.remove(oldest_gz_log)
+            # Rename the remaining gzip'd logs...
+            arch_files = glob.iglob(base_file_pattern)
+            for archFile in arch_files:
+                fn = archFile.split('.')
+                # The gzip'd file "number" is in array element 2
+                newgz_file_num = int(fn[2]) - 1
+                newgz_file = fn[0] + "." + fn[1] + "." + str(newgz_file_num) + ".gz"
+                os.rename(archFile, newgz_file)
+        else:
+            # haven't hit the maximum backup file count yet, so simply construct the
+            # next gzip file name based on the existing base file name components
+            fn = (self.baseFilename + ".1.gz").split('.')
+
+        nextgz_file_name = fn[0] + "." + fn[1] + "." + str(nextgz_file_int) + ".gz"
+        dest = self.baseFilename + ".1"
+
+        # Start a thread to gzip the output file so as not to delay primary task(s)
+        done_flag = False
+        ex = futures.ThreadPoolExecutor(max_workers=2)
+        print('Starting GZIP thread for log file {dest}...')  # debug
+        f = ex.submit(self.gzip_log, dest, nextgz_file_name)
+        f.add_done_callback(self.done)
+
+    @staticmethod
+    def gzip_log(dest, nextgz_file_name):
+
+        global gzip_in_progress
+
+        gzip_in_progress = True
+        try:
+            with open(dest, 'rb') as f_in, gzip.open(nextgz_file_name, 'wb') as f_out:
+                f_out.writelines(f_in)
+        except Exception as e:
+            return F'Error occurred during attempt to GZIP log {dest}:  {e}', dest
+        return F'Log file {dest} successfully GZIP''d...', dest
+
+    @staticmethod
+    def done(fn):
+        global done_flag
+        global gzip_in_progress
+
+        gzip_in_progress = False        # reset the GZIP-in-progress flag
+
+        result = fn.result()
+
+        if fn.cancelled():
+            print(F"GZIP thread cancelled")
+            # The destination file is returned as the second element of the result list
+            print(F"...log file {result[1]} should be manually GZIP'd and the source deleted.")
+            logger.error(F"GZIP thread cancelled")
+            logger.error(F"...log file {result[1]} should be manually GZIP'd and the source deleted.")
+        elif fn.done():
+            error = fn.exception()
+            if error:
+                print(F"Error occurred while attempting to GZIP log file...")
+                print(F"...log file {result[1]}")
+                print(F"...error returned: {error}")
+                print(F"...log file {result[1]} should be manually GZIP'd and the source deleted.")
+                logger.error(F"Error occurred while attempting to GZIP log file...")
+                logger.error(F"...log file {result[1]}")
+                logger.error(F"...error returned: {error}")
+                logger.error(F"...log file {result[1]} should be manually GZIP'd and the source deleted.")
```

Comment:
Suggestion for these blocks:

```python
str = """Error occurred while attempting to GZIP log file...
...log file {result[1]}
...error returned {error}
...log file {result[1]} should be manually GZIP'ed log file..."""
print(str)
logger.error(str)
```

I.e., use the """ multi-line string construct in Python for convenience, and also put the contents in a variable that you can both `print()` and `logger.error()`.  That way you don't have to repeat the content twice.